arch/arm/src/lpc43xx/lpc43_cgu.c

```diff
@@ -654,7 +654,7 @@ void lpc43_clockconfig(void)
 
   /* Delay around 10 milliseconds */
 
-  up_mdelay(XTAL_DELAY);
+  nxsig_usleep(XTAL_DELAY);
```

Comment:
It is written in comment that `Delay around 10 milliseconds` and function is `lpc43_clockconfig` that is called from `__start`. Are you sure that scheduler is already running at this time?

In general I think that we need to be more careful with changing things that are working.

I will examine more carefully all other places. The first impression that we are trying to fix things that are not broken and might break the working cases.

Reply:
I agree 100% - If this PR is to come in, every change needs to be checked for where it is run from. i.e. what stage is the system in and if the delay was uninterruptable for HW timing requirements.

Reply:
@xiaoxiang781216 [this](https://github.com/apache/incubator-nuttx/pull/5493/commits/2baf39c1e23f8e72cbb7fe836f6b9c6da3d9c6cf)  type of regression  is the sort of thing we need to prevent from sneaking in in this set of changes.

Reply:
I will check each location one by one, and mark this PR is ready for review when I finish. Of course, it need more review from you.